browser/brave_vpn/dns/brave_vpn_helper_watcher_win.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_vpn/dns/brave_vpn_helper_watcher_win.h"
+
+#include <utility>
+
+#include "base/logging.h"
+#include "base/task/task_traits.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_vpn/browser/connection/win/brave_vpn_helper/scoped_sc_handle.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+
+namespace {
+
+VOID CALLBACK OnServiceStoppedCallback(PVOID pParameter) {
+  SERVICE_NOTIFY* service_notfiy =
+      reinterpret_cast<SERVICE_NOTIFY*>(pParameter);
+  if (!service_notfiy || !service_notfiy->pContext) {
+    return;
+  }
+  SetEvent(service_notfiy->pContext);
+}
+
+void WaitForEvent(HANDLE event,
+                  SC_HANDLE service,
+                  SERVICE_NOTIFY* service_notify) {
+  if (NotifyServiceStatusChange(service, SERVICE_NOTIFY_STOPPED,
```

Comment:
Q) when this method failed what happens next? We don't need to do anything here?

Reply:
Good catch, added firing event to make fallback if we are unable to subscribe for the service.

Reply:
It might be worth capturing the error code and logging if it's not `ERROR_SUCCESS`. You can get a generic message to show using FormatMessage
https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage

Two examples where it can fail (from https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-notifyservicestatuschangea) 
1. if the service has been marked for deletion
2. if the notification is too far behind. With this case the happy resolution is to close handle to SCM, create again, and try the `NotifyServiceStatusChange` call again

We don't need to change the error handling behavior - but being able to distinguish between these two cases (and then others which are possible too) would he helpful 🙂

Reply:
done